src/ast/inline-document-descriptor.ts

```diff
@@ -57,33 +57,39 @@ export class InlineParsedDocument<N> implements ScannedFeature {
 
   scannedDocument: ScannedDocument;
 
+  sourceRange: SourceRange;
+
   constructor(
       type: string, contents: string, node: N, locationOffset: LocationOffset,
-      attachedComment: string) {
+      attachedComment: string, sourceRange: SourceRange) {
     this.type = type;
     this.contents = contents;
     this.node = node;
     this.locationOffset = locationOffset;
     this.attachedComment = attachedComment;
+    this.sourceRange = sourceRange;
   }
 }
 
-export function correctSourceLocation(
-    sourceLocation: SourceLocation,
-    locationOffset?: LocationOffset): SourceLocation|undefined {
-  if (!locationOffset || !sourceLocation) {
-    return sourceLocation;
+export function correctSourceRange(
+    sourceRange: SourceRange, locationOffset?: LocationOffset): SourceRange|
+    undefined {
+  if (!locationOffset || !sourceRange) {
+    return sourceRange;
   }
-  const result: SourceLocation = {
-    line: sourceLocation.line + locationOffset.line,
-    // The location offset column only matters for the first line.
-    column: sourceLocation.column +
-        (sourceLocation.line === 0 ? locationOffset.col : 0),
+  return <SourceRange>{
```

Comment:
I don't think you want to cast here, I'm not 100% sure but I think this might silence errors where the object doesn't actually conform to SourceRange